bpf/tc/attach.go

```diff
@@ -105,6 +105,19 @@ func (ap AttachPoint) AttachProgram() error {
 		return err
 	}
 
+	matchedHash, objHash, err := bpf.CheckAttachedProgs(ap.IfaceName(), ap.FileName())
+	if err == nil {
+		for _, p := range progsToClean {
+			if matchedHash && ap.FileName() == p.object {
+				logCxt.Info("Program already attached, skip re-attaching")
+				return nil
+			}
+		}
+	} else {
+		logCxt.Info(err)
```

Comment:
`logrus` isn't meant to be used like this.  You should do
```
logCtx.WithError(err).Warn("Failed to check if BPF program was already attached. Reattaching it to make sure.")
```

Reply:
Aha, I see. Thanks.